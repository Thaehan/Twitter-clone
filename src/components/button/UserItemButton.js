import {
  View,
  Text,
  Image,
  TouchableOpacity,
  TextInput,
  StyleSheet,
} from 'react-native';

import {
  GLOBAL_STYLES,
  SCREEN_WIDTH,
} from '../../styles/Style';
import AvatarButton from './AvatarButton';

//Nhận vào style, avatar, username, fullname, onPress
export default function UserItemButton({
  style,
  onPress,
  avatar,
  size,
  fullname,
  username,
}) {
  return (
    <View style={style}>
      <TouchableOpacity
        style={styles.container}
        onPress={onPress}
      >
        <View style={styles.imageContainer}>
          <Image
            source={{ uri: avatar }}
            style={{
              height: size,
              width: size,
              borderRadius: 50,
            }}
          />
        </View>
        <View style={styles.textContainer}>
          <Text style={GLOBAL_STYLES.fullname}>
            {fullname}
          </Text>
          <Text
            style={[
              GLOBAL_STYLES.username,
              styles.username,
            ]}
          >
            @{username}
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
const styles = StyleSheet.create({
  avatar: {
    borderRadius: 50,
  },
  container: {
    flexDirection: 'row',
    height: 60,
    width: SCREEN_WIDTH,
  },
  imageContainer: {
    alignItems: 'center',
    height: 60,
    justifyContent: 'center',
    width: SCREEN_WIDTH - 340,
  },
  textContainer: {
    flexDirection: 'column',
    height: 60,
    paddingTop: 10,
    width: 340,
  },
  username: {
    marginTop: 5,
  },
});
