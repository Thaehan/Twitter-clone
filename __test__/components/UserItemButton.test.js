import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import store from '../../src/redux/store';
import UserItemButton from '../../src/components/button/UserItemButton';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<UserItemButton />', () => {
  it('Render UserItemButton', () => {
    const testComponent = render(<UserItemButton />);
    expect(testComponent.toJSON()).toMatchSnapshot();
    // testComponent.getByTestId('AvatarImage').tobe();
  });
});
