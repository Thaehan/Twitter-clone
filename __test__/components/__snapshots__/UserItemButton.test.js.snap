// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItemButton /> Render UserItemButton 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": 60,
        "opacity": 1,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "width": 410,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 50,
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "height": 60,
          "paddingTop": 10,
          "width": 340,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#86939A",
              "fontSize": 15,
              "fontWeight": "400",
            },
            Object {
              "marginTop": 5,
            },
          ]
        }
      >
        @
      </Text>
    </View>
  </View>
</View>
`;
